Ginger/Ginger/ALM/ALMItemsFieldsConfigurationPage.xaml.cs

```diff
@@ -88,12 +88,15 @@ public void ShowAsWindow(eWindowShowStyle windowStyle = eWindowShowStyle.Dialog)
 
         private void Save(object sender, RoutedEventArgs e)
         {
+            // TODO Rearrange save function to keep old fields value.
             ObservableList<ExternalItemFieldBase> tempItemList = new ObservableList<ExternalItemFieldBase>();
-            foreach (ExternalItemFieldBase field in mItemsFields.Where(x => x.ToUpdate == true).ToList())
-                tempItemList.Add(field);
-            App.UserProfile.Solution.ExternalItemsFields = tempItemList;
+            //foreach (ExternalItemFieldBase field in mItemsFields.Where(x => x.ToUpdate == true).ToList())
+            //    tempItemList.Add(field);
+            App.UserProfile.Solution.ExternalItemsFields = ALMIntegration.Instance.getUpdatedFields(mItemsFields, false);
             App.UserProfile.Solution.SaveSolution(true, SolutionGeneral.Solution.eSolutionItemToSave.ALMSettings);
+            App.UserProfile.Solution.ExternalItemsFields = mItemsFields;
             genWin.Close();
```

Comment:
why need to set Solution.ExternalItemsFields = mItemsFields after save? it loses it in save process?

Reply:
sending for save only the "changed from default" and after save set all fields to cache, i will change it while jira development, for now i just improve the old code.